src/manifest/dbtProject.ts

```diff
@@ -958,7 +959,54 @@ select * from renamed
     }
     this.queryResultPanel.getSummary(compiledSql, this.adapterType);
   }
+  private getLimitQuery(queryTemplate: string, query: string, limit: number) {
+    return queryTemplate
+      .replace("{query}", () => query)
+      .replace("{limit}", () => limit.toString());
+  }
+  private async getQuery(
```

Comment:
I would prefer these helper functions to be extracted to separate file. @mdesmet your thoughts?

Reply:
Just moving these helper functions into a separate file benefit us how? If we want to refactor this, it should be rather something about making query execution easier. Maybe moving all code related to query execution into a single class with a clear API and make it a collaborator of the DbtProject could work but let's keep this refactoring out of this PR.